snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotId.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.snapshots.impl;
+
+import io.camunda.zeebe.snapshots.SnapshotId;
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class FileBasedSnapshotId implements SnapshotId {
```

Comment:
:wrench: Could be a record I think.

Reply:
Yes. But it will change the Interface methods and would result it lot of changes. Since this PR is already too big, and this is unrelated to the main issue I wouldn't do it now. I will open another PR.